deps/spidershim/test/value.cc

```diff
@@ -778,6 +778,24 @@ TEST(SpiderShim, Date) {
   EXPECT_EQ(time, Date::Cast(*date.ToLocalChecked()->ToObject())->ValueOf());
 }
 
+namespace {
+
+class TestExternalStringResource : public String::ExternalStringResource {
+  public:
+    TestExternalStringResource(const uint16_t* source, size_t length)
+        : data_(source), length_(length) {}
+    ~TestExternalStringResource() {
+      // XXX Check that we get called when the string is finalized.
```

Comment:
The v8 tests have a trick to test this by passing a reference to a counter and store it in the object, and increment the counter in the destructor, and forcing a GC and checking the counter value.  We don't have anything exposed to force a SpiderMonkey GC yet and I prefer to keep it that way if we can.  However, you can pass in a pointer to a global `bool` or something like that, and add an `atexit` callback that checks whether that `bool` has been set as a result of this finalizer being called.  That would be some dark magic but I think it should work reliably.